### PR TITLE
New NumFOCUS badge mockup

### PR DESCRIPTION
**Pull request by sriramsk1999**
So after an IRC discussion, I've added the NumFOCUS badge to the mlpack page along with the code coverage and build status badges. 

Obviously, I'm biased in thinking that this looks good so let me know what all of you think :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ src="https://cdn.rawgit.com/mlpack/mlpack.org/e7d36ed8/mlpack-black.svg" style="
   <a href="http://ci.mlpack.org/job/mlpack%20-%20git%20commit%20test/"><img src="https://img.shields.io/jenkins/build.svg?jobUrl=http%3A%2F%2Fci.mlpack.org%2Fjob%2Fmlpack%2520-%2520git%2520commit%2520test%2F&label=Linux%20build&style=flat-square" alt="Jenkins"></a>
   <a href="https://coveralls.io/github/mlpack/mlpack?branch=master"><img src="https://img.shields.io/coveralls/mlpack/mlpack/master.svg?style=flat-square" alt="Coveralls"></a>
   <a href="https://opensource.org/licenses/BSD-3-Clause"><img src="https://img.shields.io/badge/License-BSD%203--Clause-blue.svg?style=flat-square" alt="License"></a>
+  <a href="http://numfocus.org"><img src="https://img.shields.io/badge/powered%20by-NumFOCUS-orange.svg?style=flat&colorA=E1523D&colorB=007D8A" alt="NumFOCUS"></a>
```
Comment (zoq) [inline README.md]: Not sure powered by is a good wording here, also, this doesn't use the flat design as used by the other badges.

Comment (sriramsk1999) [reply README.md]: What do you suggest instead? Sponsored by?

Comment (birm) [reply README.md]: If we want to follow the wording at the bottom of https://numfocus.org/project/mlpack then I guess we'd go with "Fiscally Supported Project". That seems pretty clunky though. Maybe "Sponsored Project" to match the menus? Can we write "Affiliated" without being wrong?

Comment (sriramsk1999) [reply README.md]: I agree "Fiscally Supported Project" is a bit wordy. I'd prefer something short and sweet for the badge. 

Powered by NumFOCUS
Sponsored by NumFOCUS
Affliated to NumFOCUS (not sure if it is right or not)

These would be my picks.

Comment (rcurtin) [reply README.md]: I'd vote for `Sponsored by` :+1:

File: README.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ src="https://cdn.rawgit.com/mlpack/mlpack.org/e7d36ed8/mlpack-black.svg" style="
   <a href="http://ci.mlpack.org/job/mlpack%20-%20git%20commit%20test/"><img src="https://img.shields.io/jenkins/build.svg?jobUrl=http%3A%2F%2Fci.mlpack.org%2Fjob%2Fmlpack%2520-%2520git%2520commit%2520test%2F&label=Linux%20build&style=flat-square" alt="Jenkins"></a>
   <a href="https://coveralls.io/github/mlpack/mlpack?branch=master"><img src="https://img.shields.io/coveralls/mlpack/mlpack/master.svg?style=flat-square" alt="Coveralls"></a>
   <a href="https://opensource.org/licenses/BSD-3-Clause"><img src="https://img.shields.io/badge/License-BSD%203--Clause-blue.svg?style=flat-square" alt="License"></a>
+  <a href="http://numfocus.org"><img src="https://img.shields.io/badge/powered%20by-NumFOCUS-orange.svg?style=flat-square" alt="NumFOCUS"></a>
```
Comment (zoq) [inline README.md]: Do you mind to use the same colors as before, because those are the NumFOCUS colors.

File: README.md (hunk 1 above)
Comment (zoq) [reply README.md]: I don't really have a preference, I wonder if we really want to add that badge, I think it somewhat sticks out. However, if the mass likes the look fine with me as well.

**Comment by sriramsk1999** [comment]
I've made the requested changes. Only thing left to decide is if the badge is worth adding :) Oh and if the image banner after the second paragraph should be kept.

**Comment by sriramsk1999** [comment]
You're right, the badge should definitely link to mlpack's donation page. As for the logo, I also think it should point to NumFOCUS's main page (if we are keeping it that is.)

**Comment by rcurtin** [comment]
Personally, my thought is that if we have the button like you've added here, we can remove the bits from the README that we are a fiscally sponsored project.  Those details can be pointed out on the website (see mlpack/mlpack.org#13 and mlpack/mlpack.org#15).

Overall my take is that it would be an improvement, so I'd like to see it merged, but my opinion isn't too strong here.

**Comment by zoq** [comment]
> Personally, my thought is that if we have the button like you've added here, we can remove the bits from the README that we are a fiscally sponsored project. Those details can be pointed out on the website (see [mlpack/mlpack.org#13](https://github.com/mlpack/mlpack.org/issues/13) and [mlpack/mlpack.org#15](https://github.com/mlpack/mlpack.org/issues/15)).

If I remember right, they ask us to use https://github.com/numfocus/templates in the onbording, not sure why Shōgun removed that part from there Readme.

**Comment by sriramsk1999** [comment]
@rcurtin : I don't think it's possible to have a strong opinion on this trivial a change :) 
I've gone ahead and made the changes you've mentioned. 

PS ; [Grip](https://github.com/joeyespo/grip) is great for testing markdown.

**Comment by sriramsk1999** [comment]
@zoq : Ah didn't notice your comment :) I'll go ahead and revert it.

**Comment by sriramsk1999** [comment]
Right, reverted the changes and added the fiscal sponsor attribution mentioned in that link. Appears to have been missed the first time. Or perhaps there's some reason I'm unaware of.

**Comment by sriramsk1999** [comment]
> This may sound silly, but if we _are_ going to have a badge and the numfocus link, wouldn't it make more sense to switch what they link to? The numfocus logo to the numfocus site, and the badge to the mlpack donation page?

I'd forgotten to change where the logo points to, just made that change. Hope that's okay.

**Comment by rcurtin** [comment]
Works for me too. I forgot that we had to keep the NumFOCUS blurb because that's specifically what they asked for; thanks @zoq for pointing that out. :+1:
